### ISSUE TITLE
Reading in distances from GSSI data

### ISSUE THREAD

**Issue by christophwelling** [dlilien/ImpDAR#40]
Hello,
I am trying to use ImpDAR to read in some GSSI data we took with a ground penetrating radar. Instead of using a GPS (which would not be precise enough for us), we marked points at known intervals along the survey path, marked when we passed them during our survey and then used Radan 7's "distance normalization" feature to calculate the distance alon the survey path for each trace.
I would like to have this information available when analyzing the data in ImpDAR, but right now this distance information is only read from the GPS data. Is there a way to get ImpDAR to read the distances directly from the .dzt files, if it is available?
I would be happy to help implement this feature, but I do not understand the structure if those .dzt file. So I would really appreciate if someone could help me.

**Comment by dlilien** [comment]
Which GSSI controller, and what information did you enter on it? Is this just like a flag for different trace numbers, or is there geospatial information that was given to the radar controller?

The DZT format was documented a bit along with the old SIR 2000 if I recall correctly, which is what I used to write the loader in ImpDAR. I think there are still comments in the load_gssi.py script that should have a good bit of info on the header. Perhaps check those and see which are non-empty.

I can probably help, but I feel like I am missing something. I have dealt with something similar in the past for the opposite reason (no need to be too precise, just mark the start/end points), and unless you had an INS or this is a really weird survey I do not see how you can beat GPS precision here (for example, how were you controlling your speed and travel direction during the survey in a way that was more precise than GPS?).

**Comment by christophwelling** [comment]
Thanks for replying so quickly!
The we are using the HS200 GPR from GSSI.
We are setting flags on certain trace numbers during the survey, which are then converted into distances along the survey path by Radan 7's "distance normalization feature". So I would like to read in either these flags or the distances that Radan calculated.

Our survey is indeed pretty weird. I am working on the Radio Neutrino Observatory in Greenland (here's our whitepaper: https://arxiv.org/abs/2010.12279) and we are using the GPR to determine the positions of our antennas (as they might move a bit during deployment) and measure the index of refraction of the firn. So we are only doing GPR surveys over distances of a few tens of meters. GPS precision is also pretty bad in Greenland, so the internal GPS system of the GPR has pretty large uncertainties. We are thinking of strapping a survey GPS that lets us to PPK corections with a base station to the GPR, but we a re not quite sure how well this would work.

**Comment by christophwelling** [comment]
I've been looking for the documentation for the SIR2000 but couldn't find it. Could you point me to where it is?

**Comment by dlilien** [comment]
Yeah, typically what we do in remote areas is to use the internal GPS to get an accurate time value (the time is good even if the position is not precise) and then pair with a dual-frequency GPS mounted on the antenna and a separate base station. Set up properly, that should be much more precise than waypoints.

I don't have a copy of Radan, so I don't know what that feature looks like, but I think this should all be easy if you can read in the flags.

For the documentation, maybe it wasnt the 2000. Try page 105 here: https://www.geophysical.com/wp-content/uploads/2017/10/GSSI-SIR-30-Manual.pdf
